adr/ADR-44.md

```diff
@@ -177,8 +182,31 @@ the meta leader version is known - but in reality one cannot really tell a lot f
 guaranteed to be representative of the cluster and is particularly problematic in long running clients as the server
 may have been upgraded since last `INFO` call.
 
-In the future we could have clients assert that a certain API call requires a certain API Level but it was felt that
-today that would not be feasible to do given the amount of clients and their current state.
+# Required API level
+
+Clients can assert that a certain API call requires a minimum API Level by including `RequiredApiLevel` in stream and
+consumer create and update requests.
```

Comment:
I believe we'd have to have more things covered by this - more or less all things.

Given that we have direct gets behaving differently on the presence of headers/keys in JSON etc, we would probably even need it there so that clients can safely determine if a feature is supported.

This is why the related issue and comments is calling out a) all api requests b) a header since not all things take structs and we probably wont want to update every single structure